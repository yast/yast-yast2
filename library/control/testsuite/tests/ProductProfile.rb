# encoding: utf-8

#  ProductProfile.ycp
#  Tests of ProductProfile routines
module Yast
  class ProductProfileClient < Client
    def main
      # testedfiles: ProductProfile.ycp
      Yast.import "Testsuite"
      Yast.import "ProductProfile"

      @READ = { "target" => { "tmpdir" => "/tmp/YaST" } }
      @EX = { "target" => { "bash_output" => {} } }

      # just returns true because of non-installation mode
      Testsuite.Test(lambda { ProductProfile.CheckCompliance(nil) }, [
        @READ,
        {},
        @EX
      ], 0)

      Testsuite.Test(lambda { ProductProfile.compliance_checked }, [], 0)

      Yast.import "Mode"
      Mode.SetMode("installation")

      ProductProfile.compliance_checked = { 1 => true }

      # exits on compliance_checked test
      Testsuite.Test(lambda { ProductProfile.CheckCompliance(nil) }, [
        @READ,
        {},
        @EX
      ], 0)

      Testsuite.Test(lambda { ProductProfile.CheckCompliance(1) }, [
        @READ,
        {},
        @EX
      ], 0) 

      # this would continue to IsCompliant and initialize Pkg::
      # Testsuite::Test (``(ProductProfile::CheckCompliance (2)), [ READ, $[], EX ], 0);

      nil
    end
  end
end

Yast::ProductProfileClient.new.main
