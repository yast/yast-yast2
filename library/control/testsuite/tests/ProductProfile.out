Return	true
Return	$[]
Return	true
Return	true
